crate/interfaces/src/hsm/hsm_store.rs

```diff
@@ -236,9 +237,178 @@ impl ObjectsStore for HsmStore {
         user_must_be_owner: bool,
         params: Option<Arc<dyn SessionParams>>,
     ) -> InterfaceResult<Vec<(String, State, Attributes)>> {
-        // HSM Objects cannot be searched
-        Ok(vec![])
+        let slot_ids = self.hsm.get_available_slot_list().await?;
+        let mut uids = Vec::new();
+        if user_must_be_owner && user != self.hsm_admin {
+            return Ok(uids);
+        }
+
+        let search_attributes =
+            researched_attributes.map_or_else(Attributes::default, |search_attributes| {
+                debug!("Search attributes: {:?}", search_attributes);
+                search_attributes.clone()
+            });
+
+        if !check_basic_compatibility(&search_attributes, state) {
+            return Ok(uids);
+        }
+        let Some(object_filter) = derive_hsm_object_filter(&search_attributes) else {
+            return Ok(uids);
+        };
```

Comment:
Add `impl TryFrom<&Attributes> for HsmObjectFilter` and replace this by:

```suggestion
        let object_filter = HsmObjectFilter::try_from(&search_attributes)?;
```